### PR TITLE
Feature/localisation debugging

### PR DESCRIPTION
**Pull request by simonmitchell**
## Description

Adds a debug tool to storm section of Baymax which allows debugging localised strings in-app! This pre-pends the human readable string with the key in square brackets: "[_SOME_LOCALISATION_KEY] Hello World"

## Motivation and Context

This allows clients and PMs (And us) to tell with more ease whether a certain string is hard-coded or localised properly

## How Has This Been Tested?

Has been tested locally within Blood project

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: ThunderCloud/LocalisationsTableViewController.swift [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+//  Copyright © 2020 threesidedcube. All rights reserved.
+//
+
+import UIKit
+import ThunderTable
+
+class LocalisationsTableViewController: TableViewController {
+    
+    override func viewDidLoad() {
+
```
Comment (BenCShutt) [inline ThunderCloud/LocalisationsTableViewController.swift:15]: We need to have a word about you adding a space from the function to the `super` call! 🤣

File: ThunderCloud/LocalisedString.swift [hunk 2]
```diff
@@ -44,6 +44,9 @@ public extension String {
     func localised(with key: String, paramDictionary: [String: Any]?) -> String {
         var string = NSAttributedString(string: self).localised(with: key, paramDictionary: paramDictionary).string
         string.localisationKey = key
+        if LocalisationsTool.showDebugLocalisations && !string.hasPrefix("[\(key)]") {
```
Comment (BenCShutt) [inline ThunderCloud/LocalisedString.swift:47]: @simonmitchell This code will get hit A LOT and I'm a little worried about this constantly checking the `UserDefaults`. I _thiiiiink_ the `UserDefaults` is `Dictionary` in memory which updates a file when mutated. But if there are any IO (or DB) operations at all we need to cache this flag in memory ourselves.

Comment (BenCShutt) [inline ThunderCloud/LocalisedString.swift:47]: Also, feel like I'm missing something obvious, but when the prefix check?

File: ThunderCloud/LocalisationsTableViewController.swift (hunk 1 above)
Comment (simonmitchell) [reply ThunderCloud/LocalisationsTableViewController.swift:15]: I have a habit, and not sure where it's come from exactly... of always including a white-space line if a method is over a certain size :P See the whole of this file... yet in `LocalisationsTool.swift` I've not put a white-space line in `launchUI(in:)`! Is there anything in the swift style guidelines about it?

**Review by BenCShutt — CHANGES_REQUESTED** [review]

File: ThunderCloud/LocalisedString.swift (hunk 2 above)
Comment (simonmitchell) [reply ThunderCloud/LocalisedString.swift:47]: I'll add in-memory caching, that's a great shout!

Comment (simonmitchell) [reply ThunderCloud/LocalisedString.swift:47]: So basically I originally added this into all 3 base string methods... It turns out they all end up calling the method on localised string... but then I also added the call to `StormLanguageController` itself... If any of that logic changes I thought it best to keep it safe so we don't break this functionality (So we still add the prefix at the call-site even if underlying methods are changed to not add it) at a future date. Happy to remove though if we wish and rely entirely on `StormLanguageController`?

File: ThunderCloud/LocalisationsTableViewController.swift (hunk 1 above)
Comment (BenCShutt) [reply ThunderCloud/LocalisationsTableViewController.swift:15]: Haha nothing I'm aware of, but maybe! 😛

File: ThunderCloud/LocalisedString.swift (hunk 2 above)
Comment (BenCShutt) [reply ThunderCloud/LocalisedString.swift:47]: IMO it's nice to do it in one place if that's possible, and others reference that. 
But I'm happy to go with what you think is best here!

File: ThunderCloud/LocalisationsTool.swift [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    var displayName: String {
+        return "Localisations"
+    }
+    
+    func launchUI(in navigationController: UINavigationController) {
+        let tableViewController = LocalisationsTableViewController(style: .grouped)
+        navigationController.show(tableViewController, sender: nil)
+    }
+    
+    private static var _showDebugLocalisations: Bool?
```
Comment (BenCShutt) [inline ThunderCloud/LocalisationsTool.swift:24]: Nice, that protects us 😄

**Review by BenCShutt — APPROVED** [review]

File: ThunderCloud/LocalisedString.swift (hunk 2 above)
Comment (simonmitchell) [reply ThunderCloud/LocalisedString.swift:47]: Let's leave it for now, can always clean up at a later time if we reckon this is unlikely to change :)

**Pull request merged**